homeassistant/components/hdmi_cec.py

```diff
@@ -30,92 +113,307 @@
+        key = DOMAIN + '.' + device.name
+        hass.data[key] = device
+        discovery.load_platform(hass, base_config.get(core.DOMAIN).get(
+            CONF_CUSTOMIZE, {}).get(key, {}).get(CONF_PLATFORM, platform),
+                                DOMAIN, discovered={ATTR_NEW: [key]},
+                                hass_config=base_config)
+
+    def _shutdown(call):
+        hdmi_network.stop()
 
     def _start_cec(event):
-        """Open CEC adapter."""
-        adapters = _CEC.DetectAdapters()
-        if len(adapters) == 0:
-            _LOGGER.error("No CEC adapter found")
-            return
+        """Register services and start HDMI network to watch for devices."""
+        descriptions = load_yaml_config_file(
+            os.path.join(os.path.dirname(__file__), 'services.yaml'))[DOMAIN]
+        hass.services.register(DOMAIN, SERVICE_SEND_COMMAND, _tx,
+                               descriptions[SERVICE_SEND_COMMAND],
+                               SERVICE_SEND_COMMAND_SCHEMA)
+        hass.services.register(DOMAIN, SERVICE_VOLUME, _volume,
+                               descriptions[SERVICE_VOLUME],
+                               SERVICE_VOLUME_SCHEMA)
+        hass.services.register(DOMAIN, SERVICE_UPDATE_DEVICES, _update,
+                               descriptions[SERVICE_UPDATE_DEVICES],
+                               SERVICE_UPDATE_DEVICES_SCHEMA)
+        hass.services.register(DOMAIN, SERVICE_POWER_ON, _power_on)
+        hass.services.register(DOMAIN, SERVICE_STANDBY, _standby)
+        hass.services.register(DOMAIN, SERVICE_SELECT_DEVICE, _select_device)
 
-        if _CEC.Open(adapters[0].strComName):
-            hass.services.register(DOMAIN, SERVICE_POWER_ON, _power_on)
-            hass.services.register(DOMAIN, SERVICE_STANDBY, _standby)
-            hass.services.register(DOMAIN, SERVICE_SELECT_DEVICE,
-                                   _select_device)
-        else:
-            _LOGGER.error("Failed to open adapter")
+        hdmi_network.set_new_device_callback(_new_device)
+        hdmi_network.start()
 
     hass.bus.listen_once(EVENT_HOMEASSISTANT_START, _start_cec)
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, _shutdown)
     return True
+
+
+class CecDevice(Entity):
+    """Representation of a HDMI CEC device entity."""
+
+    def __init__(self, hass: HomeAssistant, device, logical):
+        """Initialize the device."""
+        self._device = device
+        self.hass = hass
+        self._icon = None
+        self._state = STATE_UNKNOWN
+        self._logical_address = logical
+        self.entity_id = "%s.%d" % (DOMAIN, self._logical_address)
+        device.set_update_callback(self._update)
+
+    def update(self):
+        """Update device status."""
+        self._update()
+
+    def _update(self, device=None):
+        """Update device status."""
+        if device:
+            from pycec.const import STATUS_PLAY
```

Comment:
Please group these imports. If you add a newline immediately after opening a bracket you have lots of space for the constant names, i.e.:

```python
from pycec.const import (
    STATUS_PLAY, STATUS_STOP, POWER_OFF, POWER_ON)
```